### PR TITLE
feat: [CDS-52371]: Add terraform cloud rollback step

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
#### Screenshots

Screenshot 2023-02-22 at 6 07 27 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by rameshx — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  formikRef: StepFormikFowardRef<TerraformCloudRollbackData>
+): React.ReactElement {
+  const { initialValues, onUpdate, isNewStep, readonly, allowableTypes, stepViewType, onChange } = props
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  return (
+    <>
+      <Formik<TerraformCloudRollbackData>
+        onSubmit={(values: TerraformCloudRollbackData) => {
+          onUpdate?.(values)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:78]: We can update this naming. onSubmit?

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+        }}
+        validate={(values: TerraformCloudRollbackData) => {
+          onChange?.(values)
+        }}
+        formName="terraformCloudRollback"
+        initialValues={initialValues}
+        validationSchema={Yup.object().shape({
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:85]: We can move this validation in some const and use here. that will clear the clutter here in the element.

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+  stepViewType
+}) => {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  const prefix = isEmpty(path) ? '' : `${path}.`
+  return (
+    <>
+      {isValueRuntimeInput(template?.timeout) && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <FormMultiTypeDurationField
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:215]:
```suggestion
          < TimeoutFieldInputSetView
```
To support allowed values for timeout field.

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+              disabled: readonly
+            }}
+            label={getString('pipelineSteps.timeoutLabel')}
+            name={`${prefix}timeout`}
+            disabled={readonly}
+          />
+        </div>
+      )}
+      {isValueRuntimeInput(template?.spec?.message) && (
+        <FormMultiTypeTextArea
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:231]:
```suggestion
<div className={cx(stepCss.formGroup, stepCss.md)}>
        <FormMultiTypeTextArea
```
Is the wrapper classes not need here?

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx (hunk 4 above)
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:231]: No, this is as per mocks

File: src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx (hunk 3 above)
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx:215]: sure, will update in next PR of terraform cloud run step

**Pull request merged**